src/Rector/Deprecation/Base/DBBase.php

```diff
@@ -49,6 +50,18 @@ abstract class DBBase extends AbstractRector
      */
     protected $optionsArgumentPosition;
 
+    private $parameterProvider;
+
+    public function __construct(ParameterProvider $parameterProvider)
+    {
+        $this->parameterProvider = $parameterProvider;
+    }
+
+    protected function getParameterProvider(): ParameterProvider
+    {
+        return $this->parameterProvider;
+    }
+
```

Comment:
This is a lot of boilerplate and should really go into a shared service that all of the rules use. However, we delegate all ownership of configuration to end-users. So, this is a bit messy but there's no good workaround since the property was removed from AbstractRector base class.

This is used to check if `drupal_rector_notices_as_comments` has been flagged in the configuration:

```
$parameters->set('drupal_rector_notices_as_comments', true);
```

Part of me wonders if anyone ever sets that to false and if we should just default to `true` and add comments.

Reply:
Resolved by using ConfigurableRectorInterface and container parameter injection.